app/controllers/pets_controller.rb

```diff
@@ -31,6 +36,7 @@ def update
 
   def destroy
     Pet.destroy(params[:id])
+    favorite.contents.delete(params[:id])
```

Comment:
this removes a pet from favorites if the pet is deleted

Reply:
Awesome addition.